protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -284,9 +289,16 @@ func (rpccs *RPCConsumerServer) SendRelay(
 	retries := uint64(0)
 	timeouts := 0
 	unwantedProviders := rpccs.GetInitialUnwantedProviders(directiveHeaders)
+
+	forceCacheRefresh := false
```

Comment:
should be in rpccs.HandleDirectiveHeadersForMessage(chainMessage, directiveHeaders)